src/Meshes/interval.jl

```diff
@@ -167,34 +191,50 @@ function IntervalMesh(
     domain::IntervalDomain{CT},
     stretch::GeneralizedExponentialStretching{FT};
     nelems::Int,
+    reverse_mode::Bool = false,
 ) where {CT <: Geometry.Abstract1DPoint{FT}} where {FT}
     if nelems ≤ 1
         throw(ArgumentError("`nelems` must be ≥ 2"))
     end
+
     dz_bottom, dz_top = stretch.dz_bottom, stretch.dz_top
-    if !(dz_bottom ≤ dz_top)
+    if !(dz_bottom ≤ dz_top) && reverse_mode == false
```

Comment:
```suggestion
    if !(dz_bottom ≤ dz_top) && !reverse_mode
```
I think this reads a bit easier

Reply:
It's subjective. For me that I have vision problem the version spelled out is easier to read than the one with the bang that could be easily missed at a first glance. But it's ok. I changed it as suggested in [0bf0a5a](https://github.com/CliMA/ClimaCore.jl/pull/1001/commits/0bf0a5ac9a4bc648a7796b7efd5465fe2d2beebf)